frontend/apps/reader/modules/readertoc.lua

```diff
@@ -436,21 +447,48 @@ function ReaderToc:isChapterEnd(cur_pageno)
 end
 
 function ReaderToc:getChapterPagesLeft(pageno)
-    --if self:isChapterEnd(pageno) then return 0 end
-    local next_chapter = self:getNextChapter(pageno)
-    if next_chapter then
-        next_chapter = next_chapter - pageno - 1
+    if self.ui.document:hasHiddenFlows() then
+        local flow = self.ui.document:getPageFlow(pageno)
+        -- Count pages until new chapter
+        local pages_left = 0
+        local test_page = self.ui.document:getNextPage(pageno)
+        while test_page > 0 do
+            pages_left = pages_left + 1
+            if self:isChapterStart(test_page) then
+                return pages_left - 1
+            end
+            test_page = self.ui.document:getNextPage(test_page)
+        end
+    else
+        local next_chapter = self:getNextChapter(pageno)
+        if next_chapter then
+            next_chapter = next_chapter - pageno - 1
+        end
+        return next_chapter
     end
-    return next_chapter
 end
 
 function ReaderToc:getChapterPagesDone(pageno)
-    if self:isChapterStart(pageno) then return 0 end
-    local previous_chapter = self:getPreviousChapter(pageno)
-    if previous_chapter then
-        previous_chapter = pageno - previous_chapter
+    if self.ui.document:hasHiddenFlows() then
+        if self:isChapterStart(pageno) then return 0 end
```

Comment:
You forgot to make the non-hasHiddenFlows branch benefit from `if self:isChapterStart(pageno) then return 0 end` - so I guess you can probably move it back at the start of the function.

getChapterPagesDone seems not used anywhere -  but I use it in my own patch to tweak the output of the `pages_left` item - so, I got wrong info.
That's my tweak:
```diff
     pages_left = function(footer)
         local symbol_type = footer.settings.item_prefix or "icons"
         local prefix = symbol_prefix[symbol_type].pages_left
         local left = footer.ui.toc:getChapterPagesLeft(footer.pageno)
-         return prefix .. " " .. (left and left or footer.pages - footer.pageno)
+        -- so we see better where we are in that chapter
+        local done = footer.ui.toc:getChapterPagesDone(footer.pageno)
+        left = left or (footer.pages - footer.pageno)
+        done = done or footer.pageno-1
+        return (left+1).." > "..(left+done+1)
```